In [1]:
import plotly
import plotly.express as px
import pandas as pd
import math

In [2]:
# Lendo o arquivo
regiao = pd.read_excel("TS_REGIAO.xlsx")
regiao.head(20)

CO_REGIAO NO_REGIAO DEPENDENCIA_ADM LOCALIZACAO   CAPITAL  MEDIA_2_LP  \
0           1     Norte        Estadual       Rural   Capital      712.03   
1           1     Norte        Estadual       Rural  Interior      719.78   
2           1     Norte        Estadual       Rural     Total      717.60   
3           1     Norte        Estadual       Total   Capital      733.81   
4           1     Norte        Estadual       Total  Interior      736.08   
5           1     Norte        Estadual       Total     Total      734.83   
6           1     Norte        Estadual      Urbana   Capital      734.81   
7           1     Norte        Estadual      Urbana  Interior      738.71   
8           1     Norte        Estadual      Urbana     Total      736.45   
9           1     Norte         Federal       Rural   Capital         NaN   
10          1     Norte         Federal       Rural  Interior         NaN   
11          1     Norte         Federal       Rural     Total         NaN   
12          1     Norte         Federal       Total   Capital      764.89   
13          1     Norte         Federal       Total  Interior         NaN   
14          1     Norte         Federal       Total     Total      764.89   
15          1     Norte         Federal      Urbana   Capital      764.89   
16          1     Norte         Federal      Urbana  Interior         NaN   
17          1     Norte         Federal      Urbana     Total      764.89   
18          1     Norte       Municipal       Rural   Capital      725.52   
19          1     Norte       Municipal       Rural  Interior      717.30   

    MEDIA_2_MT  MEDIA_5_LP  MEDIA_5_MT  MEDIA_9_LP  ...  nivel_1_MTEM  \
0       721.96      184.98      194.85      234.08  ...         19.69   
1       726.00      167.88      182.51      222.13  ...         19.36   
2       724.86      172.16      185.60      223.67  ...         19.38   
3       740.31      210.17      218.95      249.86  ...         17.21   
4       746.74      199.10      211.86      242.06  ...         18.69   
5       743.18      204.46      215.30      245.11  ...         18.23   
6       741.15      211.26      220.00      250.30  ...         17.17   
7       750.08      203.26      215.78      244.70  ...         18.57   
8       744.91      207.31      217.91      247.02  ...         18.09   
9          NaN         NaN         NaN         NaN  ...          2.84   
10         NaN         NaN         NaN         NaN  ...          9.47   
11         NaN         NaN         NaN         NaN  ...          9.06   
12      758.83      235.49      241.93      298.55  ...          3.00   
13         NaN         NaN         NaN         NaN  ...          8.32   
14      758.83      235.49      241.93      298.55  ...          6.32   
15      758.83      235.49      241.93      298.55  ...          3.00   
16         NaN         NaN         NaN         NaN  ...          8.12   
17      758.83      235.49      241.93      298.55  ...          6.03   
18      723.14      199.49      213.01      246.91  ...           NaN   
19      732.72      167.68      182.37      223.45  ...           NaN   

    nivel_2_MTEM  nivel_3_MTEM  nivel_4_MTEM  nivel_5_MTEM  nivel_6_MTEM  \
0          15.10         13.31          6.37          1.85          1.37   
1          14.92          9.19          3.85          1.31          0.41   
2          14.93          9.38          3.97          1.33          0.45   
3          18.55         16.35         10.42          5.16          1.99   
4          18.52         14.62          8.37          3.84          1.29   
5          18.53         15.15          9.00          4.25          1.51   
6          18.60         16.40         10.48          5.21          2.00   
7          19.15         15.58          9.17          4.29          1.45   
8          18.96         15.86          9.61          4.61          1.64   
9          11.78          8.94         28.06         24.39         12.20   
10         14.47

In [3]:
# Verificando o tamanho 
regiao.shape

(306, 156)

In [4]:
# Verificando a existência de valores ausentes
regiao.isnull().sum()

CO_REGIAO           0
NO_REGIAO           0
DEPENDENCIA_ADM     0
LOCALIZACAO         0
CAPITAL             0
                   ..
nivel_6_MTEM       18
nivel_7_MTEM       18
nivel_8_MTEM       18
nivel_9_MTEM       18
nivel_10_MTEM      18
Length: 156, dtype: int64

In [5]:
# Removendo todas as linhas que possuem valores ausentes (NaN)
regiao.dropna(axis=0, inplace=True)

In [6]:
# Removendo valores duplicados
regiao.drop_duplicates(keep='first', inplace=True)

In [7]:
regiao.head(20)

CO_REGIAO NO_REGIAO               DEPENDENCIA_ADM LOCALIZACAO   CAPITAL  \
1           1     Norte                      Estadual       Rural  Interior   
2           1     Norte                      Estadual       Rural     Total   
3           1     Norte                      Estadual       Total   Capital   
4           1     Norte                      Estadual       Total  Interior   
5           1     Norte                      Estadual       Total     Total   
6           1     Norte                      Estadual      Urbana   Capital   
7           1     Norte                      Estadual      Urbana  Interior   
8           1     Norte                      Estadual      Urbana     Total   
12          1     Norte                       Federal       Total   Capital   
14          1     Norte                       Federal       Total     Total   
15          1     Norte                       Federal      Urbana   Capital   
17          1     Norte                       Federal      Urbana     Total   
28          1     Norte                       Privada       Rural  Interior   
29          1     Norte                       Privada       Rural     Total   
31          1     Norte                       Privada       Total  Interior   
32          1     Norte                       Privada       Total     Total   
37          1     Norte  Total - Estadual e Municipal       Rural  Interior   
38          1     Norte  Total - Estadual e Municipal       Rural     Total   
39          1     Norte  Total - Estadual e Municipal       Total   Capital   
40          1     Norte  Total - Estadual e Municipal       Total  Interior   

    MEDIA_2_LP  MEDIA_2_MT  MEDIA_5_LP  MEDIA_5_MT  MEDIA_9_LP  ...  \
1       719.78      726.00      167.88      182.51      222.13  ...   
2       717.60      724.86      172.16      185.60      223.67  ...   
3       733.81      740.31      210.17      218.95      249.86  ...   
4       736.08      746.74      199.10      211.86      242.06  ...   
5       734.83      743.18      204.46      215.30      245.11  ...   
6       734.81      741.15      211.26      220.00      250.30  ...   
7       738.71      750.08      203.26      215.78      244.70  ...   
8       736.45      744.91      207.31      217.91      247.02  ...   
12      764.89      758.83      235.49      241.93      298.55  ...   
14      764.89      758.83      235.49      241.93      298.55  ...   
15      764.89      758.83      235.49      241.93      298.55  ...   
17      764.89      758.83      235.49      241.93      298.55  ...   
28      723.64      712.12      208.82      220.61      251.99  ...   
29      729.35      718.33      213.28      225.05      260.93  ...   
31      763.12      758.31      232.84      241.13      285.00  ...   
32      766.30      761.61      235.82      244.07      286.77  ...   
37      717.46      732.28      167.70      182.37      223.17  ...   
38      717.93      731.36      169.55      184.08      223.92  ...   
39      730.37      737.87      209.18      220.14      252.01  ...   
40      724.73      733.48      186.15      199.01      238.33  ...   

    nivel_1_MTEM  nivel_2_MTEM  nivel_3_MTEM  nivel_4_MTEM  nivel_5_MTEM  \
1          19.36         14.92          9.19          3.85          1.31   
2          19.38         14.93          9.38          3.97          1.33   
3          17.21         18.55         16.35         10.42          5.16   
4          18.69         18.52         14.62          8.37          3.84   
5          18.23         18.53         15.15          9.00          4.25   
6          17.17         18.60         16.40         10.48          5.21   
7          18.57         19.15         15.58          9.17          4.29   
8          18.09         18.96         15.86          9.61          4.61   
12          3.00          6.78         13.35         18.22         21.95   
14          6.32         11.72         17.83         18.74         18.55   
15          3.00

In [8]:
regiao_agroup = regiao.groupby(["NO_REGIAO"]).mean()
regiao_agroup.head()

CO_REGIAO  MEDIA_2_LP  MEDIA_2_MT  MEDIA_5_LP  MEDIA_5_MT  \
NO_REGIAO                                                                 
Centro-Oeste        5.0  747.860714  744.745000  213.185476  226.017857   
Nordeste            2.0  739.917292  741.524583  207.047708  218.170833   
Norte               1.0  732.848571  738.189286  200.067381  211.530000   
Sudeste             3.0  759.017400  757.657400  226.713800  241.594600   
Sul                 4.0  763.914167  758.509375  224.828958  238.803333   

              MEDIA_9_LP  MEDIA_9_MT  MEDIA_9_CH  MEDIA_9_CN  MEDIA_12_LP  \
NO_REGIAO                                                                   
Centro-Oeste  260.013095  265.429286  250.667381  254.750714   283.441667   
Nordeste      258.342292  262.384792  252.877917  247.724583   273.756042   
Norte         249.542381  251.085476  244.701667  240.132619   262.778095   
Sudeste       269.793200  275.272000  260.341600  260.478200   293.005200   
Sul           272.262708  279.028542  264.268125  266.179583   292.606042   

              ...  nivel_1_MTEM  nivel_2_MTEM  nivel_3_MTEM  nivel_4_MTEM  \
NO_REGIAO     ...                                                           
Centro-Oeste  ...     11.676905     15.853095     18.355000     16.079286   
Nordeste      ...     13.547500     15.531458     14.954375     12.018125   
Norte         ...     16.016429     16.282619     14.270238     10.298571   
Sudeste       ...      9.475400     12.807400     15.907000     15.230000   
Sul           ...      8.887500     13.145625     17.193750     17.279583   

              nivel_5_MTEM  nivel_6_MTEM  nivel_7_MTEM  nivel_8_MTEM  \
NO_REGIAO                                                              
Centro-Oeste     11.788810      6.751905      3.224048      1.167857   
Nordeste          9.317292      6.157917      3.475417      1.856458   
Norte             6.880476      3.547381      1.659286      0.803810   
Sudeste          13.479000      9.710600      6.235400      3.315400   
Sul              14.007083     10.265000      5.309167      2.994792   

              nivel_9_MTEM  nivel_10_MTEM  
NO_REGIAO                                  
Centro-Oeste      0.575000            0.0  
Nordeste          0.767292            0.0  
Norte             0.209048            0.0  
Sudeste           1.208600            0.0  
Sul               1.170417            0.0  

[5 rows x 152 columns]

In [9]:
regiao_agroup_LP = regiao_agroup['MEDIA_9_LP'].reset_index().sort_values(by='MEDIA_9_LP')
fig1 = px.bar(regiao_agroup_LP, x = 'NO_REGIAO', y = 'MEDIA_9_LP',
             labels={'NO_REGIAO': 'Regiões', 'MEDIA_9_LP': 'Notas'},
             color_discrete_sequence=px.colors.qualitative.T10, template='presentation', text='MEDIA_9_LP')
fig1.update_traces(textposition='inside',texttemplate='%{text:.5s}')
fig1.update_layout(title={'text' : 'Médias de Língua portuguesa, 9° ano em 2019', 'y': 0.95, 'x': 0.5})
fig1.show()

In [10]:
dN = regiao[regiao["NO_REGIAO"] == "Nordeste"]
dN["MEDIA_9_LP"].describe()


count     48.000000
mean     258.342292
std       23.215073
min      236.290000
25%      245.477500
50%      247.470000
75%      262.607500
max      318.760000
Name: MEDIA_9_LP, dtype: float64

In [11]:
dS = regiao[regiao["NO_REGIAO"] == "Sul"]
dS["MEDIA_9_LP"].describe()

count     48.000000
mean     272.262708
std       17.510564
min      257.400000
25%      259.397500
50%      264.105000
75%      275.070000
max      309.330000
Name: MEDIA_9_LP, dtype: float64

Teste T-Student para a Média das Notas de Português do Nordeste e Sul

Hipótese do Teste

* Hipótese Nula (H0): Não há diferença significativa entre as médias
* Hipótese Alternativa (H1): Há diferença significativa entre as médias



In [12]:
# Calculando a variância
s_variancia = (((dN["MEDIA_9_LP"].count() - 1) * pow(dN["MEDIA_9_LP"].std(), 2)) + ((dS["MEDIA_9_LP"].count() - 1) * pow(dS["MEDIA_9_LP"].std(), 2))) / ((dN["MEDIA_9_LP"].count() + dS["MEDIA_9_LP"].count()) - 2)

s_variancia

422.77972123226954

In [13]:
# Calculando o valor de t
t_calculado = (dN["MEDIA_9_LP"].mean() - dS["MEDIA_9_LP"].mean()) / math.sqrt(s_variancia * ((1/dN["MEDIA_9_LP"].count()) + (1/dS["MEDIA_9_LP"].count())))

t_calculado

-3.316658690658605

Verificar na tabela estatística de t-student o valor crítico para t

Grau de Liberdade: gl = n1 + n2 - 2     =>      gl = 48 + 48 - 2 = 94

Significancia de 0.05 (95% de certeza)  =>      t_critico = 1.9867



Se t_calculado >= t_critico rejeita H0 e aceita H1

t_calculado = |-3.32| = 3.32

t_critico = |1.9867| = 1.9867

Conclusão: Como o t_calculado é maior que o t_critico aceita-se a hipótese H1. Dessa forma, conclui-se que existe diferença estatisticamente significativa entre as notas de português do Nordeste e Sul.



In [14]:
regiao_agroup_MT = regiao_agroup['MEDIA_9_MT'].reset_index().sort_values(by= 'MEDIA_9_MT')
fig2 = px.bar(regiao_agroup_MT, x = 'NO_REGIAO', y = 'MEDIA_9_MT',
             labels={'NO_REGIAO': 'Regiões', 'MEDIA_9_MT': 'Notas'},
             color_discrete_sequence=px.colors.qualitative.T10, template='presentation', text='MEDIA_9_MT')
fig2.update_traces(textposition='inside',texttemplate='%{text:.5s}')
fig2.update_layout(title={'text' : 'Médias de Matemática, 9° ano em 2019', 'y': 0.95, 'x': 0.5})
fig2.show()

In [15]:
dN["MEDIA_9_MT"].describe()

count     48.000000
mean     262.384792
std       29.438961
min      237.860000
25%      245.977500
50%      248.400000
75%      264.857500
max      344.880000
Name: MEDIA_9_MT, dtype: float64

In [16]:
dS["MEDIA_9_MT"].describe()

count     48.000000
mean     279.028542
std       22.015220
min      259.720000
25%      265.297500
50%      267.990000
75%      279.707500
max      328.870000
Name: MEDIA_9_MT, dtype: float64

Teste T-Student para a Média das Notas de Matemática do Nordeste e Sul

Hipótese do Teste

* Hipótese Nula (H0): Não há diferença significativa entre as médias
* Hipótese Alternativa (H1): Há diferença significativa entre as médias



In [17]:
# Calculando a variância
s_variancia = (((dN["MEDIA_9_MT"].count() - 1) * pow(dN["MEDIA_9_MT"].std(), 2)) + ((dS["MEDIA_9_MT"].count() - 1) * pow(dS["MEDIA_9_MT"].std(), 2))) / ((dN["MEDIA_9_MT"].count() + dS["MEDIA_9_MT"].count()) - 2)

s_variancia

675.6611701684399

In [18]:
# Calculando o valor de t
t_calculado = (dN["MEDIA_9_MT"].mean() - dS["MEDIA_9_MT"].mean()) / math.sqrt(s_variancia * ((1/dN["MEDIA_9_MT"].count()) + (1/dS["MEDIA_9_MT"].count())))

t_calculado

-3.1368396881287843

Verificar na tabela estatística de t-student o valor crítico para t

Grau de Liberdade: gl = n1 + n2 - 2     =>      gl = 48 + 48 - 2 = 94

Significancia de 0.05 (95% de certeza)  =>      t_critico = 1.9867



Se t_calculado >= t_critico rejeita H0 e aceita H1

t_calculado = |-3.14| = 3.14

t_critico = |1.9867| = 1.9867

Conclusão: Como o t_calculado é maior que o t_critico aceita-se a hipótese H1. Dessa forma, conclui-se que existe diferença estatisticamente significativa entre as notas de português do Nordeste e Sul.